test/components/converters/test_openapi_functions.py

```diff
@@ -233,9 +233,57 @@ def test_run_with_file_source_and_none_system_messages(self, json_serperdev_open
         assert (
             doc.content
             == '{"name": "search", "description": "Search the web with Google", "parameters": {"type": "object", '
-            '"properties": {"requestBody": {"type": "object", "properties": {"q": {"type": "string"}}}}}}'
+            '"properties": {"q": {"type": "string"}}}}'
         )
 
         # check that the metadata is as expected, system_message should not be present
         assert "system_message" not in doc.meta
         assert doc.meta["spec"] == json.loads(json_serperdev_openapi_spec)
+
+    def test_greet_oas_to_openai_function(self, test_files_path):
```

Comment:
In these new unit tests, you can also see how we output the flat OpenAI structure as detailed in https://github.com/deepset-ai/haystack/pull/7034/files#r1495355996